### PR TITLE
fix: twitter address parsing

### LINKED ISSUE #2773: [BUG]  (⚡lnaddress@getalby.com) with ')' bracket following right after, activates the extension, but gives error on sending
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

:warning: [Sam](https://twitter.com/SamSethi) Sethi has/had "my Alby wallet is (:zap:[sam@getalby.com](mailto:sam@getalby.com)) and my" and while extension wants to send him sats, later it gives an error (screen in comments) - be**cause of the ")" bracket following right after**?

### Screenshots [optional]

image

### Steps To Reproduce

1. Visit page with LN address with bolt emoji in front and bracket right after. Like https://twitter.com/SamSethi
2. Chekc if extension turns blue and enables tipping
3. Try sending tip with send sats button
4. ??

### Expected behavior

It should simply send sats to that LN address and ignore the bracket or any characters

### Alby information

Alby 3.20

### Device information

Chrome, Windows

### Additional context

_No response_

### Are you working on this?

None

### PR DESCRIPTION
**Pull request by reneaaron**
### Describe the changes you have made in this PR

Fixes #2773

Test here: 
https://twitter.com/_reneaaron

### CODE REVIEW

File: src/extension/content-script/batteries/Twitter.ts [hunk 1]
```diff
@@ -109,10 +109,10 @@ function battery(): void {
         for (const zapElement of zapElements) {
           if (
             (match = (zapElement.nextSibling?.textContent || "").match(
-              /(\S+@\S+)/
+              /\b[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Za-z]{2,}\b/
```
Comment (bumi) [inline src/extension/content-script/batteries/Twitter.ts:112]: is a `%` correct?

and the `.` in `[A-Za-z0-9.-]` is probably not correct, do you mean `\.`?

theoretically there are domains with umlauts.
there subdomains and with multiple tlds (e.g. me@bitcoin.me.co.uk) - which your regex matches I think - but I don't know why :D

Comment (bumi) [reply src/extension/content-script/batteries/Twitter.ts:112]: @reneaaron ping

Comment (reneaaron) [reply src/extension/content-script/batteries/Twitter.ts:112]: Yeah, I didn't consider umlaut domains. Will try to update the regex :+1:

**Comment by reneaaron** [comment]
Closing as it's a real edge case, won't fix for now.

**Pull request closed without merging**